### ISSUE TITLE
Add endpoint to open and pre-fill "Transfer Isk" window

### ISSUE THREAD

**Issue by deveyus** [ccpgames/esi-issues#190]
Used to be able to do this with IGB.

Want to be able to open the transfer isk window and prefill the who, amount, and reason fields.

**Comment by SpeedProg** [comment]
would like this, if it includes the ability to transfer isk from corp wallet

**Comment by ddouglas493** [comment]
Any update to this?

**Comment by Greygal** [comment]
Oh please make this a thing, I beg of you! I just spent 11+ hours of my life I'll never get back manually paying out over 1,300 payments because I haven't been able to find any other way to do it ever since IGB removed.  With IGB, it would have taken me maybe 2, 3 hours at most.

**Comment by deveyus** [comment]
Yeah, I'm a little miffed admittedly, we were promised that we'd have everything the IGB could do in game BEFORE it was removed, and now about a year later, we're still waiting on one of the best uses of it.

**Comment by ddouglas493** [comment]
@deveyus They never promise to completely replace the functionality that we had in the IGB with Crest/ESI. They choose specific features and replaced them with Crest Functionality

https://community.eveonline.com/news/dev-blogs/bidding-farewell-to-the-in-game-browser/

**Comment by Dusty-Meg** [comment]
Being able to use a corp wallet if you have the roles would also be great ability.

**Comment by tsuthers-ccpgames** [comment]
There's been some discussion of this and a few other similar features. The generally settled on conclusion is that probably the best way to handle this would be to have an "API authorization queue" window in the game.

You could use the API to automatically stack up a bunch of payments, but you'd need to log into the client and hit approve (maybe approve all?) on the payments to have them pay out. This allows us to be certain of intent, and prevents someone who pinches your phone from emptying your wallet.

However, that's certainly a concept that will have to wait until after XML/Crest parity before it gets explored further.

**Comment by Dusty-Meg** [comment]
The need to see it in-game to actually press accept is something that probably should be required otherwise could be getting some rouge apps / users of apps as you said.

**Comment by deveyus** [comment]
The frustrating part is that it's functionality we had before and lost.  Would like to have some kind of solution here, we as players are willing to do a lot of the work to provide proper financial systems that aren't "One flat rate tax" for our corporation members, but we need the interfaces back to be able to do so.

**Comment by warlof** [comment]
That's could be used for replacement program feature, even if automated transaction endpoint would be prefereable, I can understand that's for security reason, IG interaction is the single option.

100% agree with the request !

**Comment by a-tal** [comment]
approved by GD

**Comment by caullerd** [comment]
Are there any plans to implement this in near future? I'm not being pushy...it's still a long awaited feature for some applications.

**Comment by deveyus** [comment]
Seeing as this was asked 2.5 years ago, it's clear it's never going to happen.  I wouldn't get your hopes up at all.

**Comment by mossyblog** [comment]
Oh man would i love this. Definitely a helpful feature for logistics etc.

**Comment by tsuthers-ccpgames** [comment]
It's predicated on having some kind of user-intent mechanism, like enforced 2FA auth or similar, to ensure that the request was made by the human who owns the account.

There are a few other concepts that are similar. Pretty much anything that involves transferring money or assets we would want some kind of proof-of-human-in-the-loop mechanism.

Thus far that mechanism doesn't exist. This would be another good topic to funnel via the CSM if there's a lot of people wanting this, since it would require engineering time from multiple teams and thus higher level support in order to move forward.

**Comment by caullerd** [comment]
Request to open prefilled window must be only issued by owner of character? Why? It's not a direct transfer request...

**Comment by tsuthers-ccpgames** [comment]
Oh true, I misread the ticket. I was referring to a direct transfer :)

**Comment by bahrmichael** [comment]
> Pretty much anything that involves transferring money or assets we would want some kind of proof-of-human-in-the-loop mechanism.
I see this as a good fit for SP extraction as well, maybe Peligro would be interested to make "rapid SP extraction" harder.

**Comment by Seavert** [comment]
IMHO.
Concept of "API authorization queue" is bad. Queue can become a mess because of injection, overflow or rewriting.
Request should contain bunch of payments and be processed independently. So you are sure that all visible payments are from single request of single application.
Here are several ideas for such API window in game.

Show proofs of bunch authenticity:
- Application name.
- Value of special request parameter: `verification_code`.

Noise protection:
- Window is locked after succesful request and further requests are ignored.
You can watch this list and have options: "Approve" separate payment, "Approve all", "Clear".
Window is unlocked if all payments are done (either way) or after "Clear".
- "Remove access" - option to remove current application. Deletes scope `send-money` (is it possible?) or all tokens (as in access web-page).

**Comment by deveyus** [comment]
I think you're all over complicating this.  It just needs to be the existing Transfer Isk window, much like the open market window, the only difference being that you include the requesting application's identifier, so you know it's coming from the right place.  If you don't want that application making requests simply remove it's access rights, same as for open market endpoint or the like. 

This isn't a request to automate the transfer of isk, merely one to allow prefilling of form fields, something we had back when we still had the IGB, proving that automation of form-filling is not something that has historically been a problem, and a feature that was lost due to the IGB being removed.

**Comment by deveyus** [comment]
It's been nearly four years? Is this something that is every going to happen, or has work on ESI simply ceased at this point?  Entire games have been developed in the time this issue has been open, it can't be that hard, and it's been approved by game design for a long time?  All we have is silence.  Opening windows is already an accepted thing ESI is allowed to do, this is functionality that once existed in game, could we at least be clued in on what's the roadblock here?  You already clearly have (or had) the hooks in the game for making calls to open this window, you have similar hooks for market, show info, and contracts; I'm simply baffled how this is still sitting here?

**Comment by winterwings** [comment]
What puzzles me is that there is even an ability to produce spam on behalf of authorised unsuspecting users (who reads the scopes in big alliances anyway?), while a simple window opener is considered bad... :(
I'm sad to see how IGB JS API was cut, but I'm happy I had an opportunity to make some useful tools for logistics back then. At least we have some memories left, right?

Anyway. My coalition needs this to ease tax payment and SRP. While SRP is currently pretty simple (send ISK to a character), tax payments gathering are not: they require to set up a contract asking for ISK, so the corporations could easily accept them, while corp director will go to the website to check the sum anyway. It'll be easier for directors to simply click the "pay" button on a website and transfer the sum, which, in turn, will ease the monthly pain of contracts creation for alliances' financial staff.

**Comment by deveyus** [comment]
Just dropping by for my yearly bump; still care about this CCP.  Still hoping you guys care about restoring lost functionality to the game.

**Comment by wcraigjones** [comment]
I'm currently building a tax system for our alliance and was surprised the ability to open the transfer window didn't exist. Now we have to teach everyone in the alliance to pay the right corp with the right reason and the right amount. All I want to do it prefil and open the UI element.
